audmath/core/api.py

```diff
@@ -86,6 +88,205 @@ def db(
     return x
 
 
+def duration_in_seconds(
+        duration: typing.Union[float, int, str, np.timedelta64],
+        sampling_rate: typing.Union[float, int] = None,
+) -> np.floating:
+    r"""Duration in seconds.
+
+    Converts the given duration value to seconds.
+    A unit can be provided
+    when ``duration`` is given as a string.
+    As units the following values are possible.
+
+    .. table::
+
+        ==================================================== ===========
+        Unit                                                 Meaning
+        ==================================================== ===========
+        W                                                    week
+        D, days, day                                         day
+        h, hours, hour, hr                                   hour
+        m, minutes, minute, min, T                           minute
+        s, seconds, second, sec, S                           second
+        ms, milliseconds, millisecond, millis, milli, L      millisecond
+        us, μs, microseconds, microsecond, micros, micro, U  microsecond
+        ns, nanoseconds, nanoseconds, nanos, nano, N         nanosecond
+        ==================================================== ===========
+
+    .. _numpy's datetime units: https://numpy.org/doc/stable/reference/arrays.datetime.html#datetime-units
+
+    Args:
+        duration: if ``duration`` is
+            a float,
+            integer
+            or string without unit
+            it is treated as seconds
+            or if ``sampling_rate`` is provided
+            as samples.
+            If ``duration`` is provided as a string with unit,
+            e.g. ``'2ms'`` or ``'2 ms'``,
+            or as a :class:`numpy.timedelta64`
+            or :class:`pandas.Timedelta` object
+            it will be converted to seconds
+            and ``sampling_rate`` is always ignored
+        sampling_rate: sampling rate in Hz.
+            Is ignored
+            if duration is provided with a unit
+
+    Returns:
+        duration in seconds
+
+    Raises:
+        ValueError: if the provided unit is not supported
+        ValueError: if ``duration`` is a string
+            that does not match a valid value unit pattern
```

Comment:
```suggestion
            that does not match a valid '<value><unit>' pattern
```

I think it's easier to understand this way.

Reply:
Was it meant like this:

![image](https://user-images.githubusercontent.com/173624/216018143-e6f1e133-ed98-48ed-98b6-53c747c7f221.png)

Or should it be presented as code?

Otherwise I would maybe remove the `'` around it.

Reply:
I think it's better to have the `'` there, then it's clear we are referring to a string. So I would leave it as is.

Reply:
ok